File: src/main/java/io/dddrive/oe/model/ObjTenant.java

package io.dddrive.oe.model;

import io.dddrive.obj.model.Obj;
import io.dddrive.oe.model.enums.CodeTenantType;

public interface ObjTenant extends Obj {

	CodeTenantType getTenantType();

	void setTenantType(CodeTenantType tenantType);

	String getName();

	void setName(String name);

	String getDescription();

	void setDescription(String description);

}

File: fm-domain/src/main/java/io/zeitwert/fm/oe/model/ObjTenant.kt
package io.zeitwert.fm.oe.model

import dddrive.app.obj.model.Obj
import io.zeitwert.fm.dms.model.ObjDocument
import io.zeitwert.fm.oe.model.enums.CodeTenantType
import java.math.BigDecimal

interface ObjTenant : Obj {

	var key: String?

	var name: String?

	var description: String?

	var tenantType: CodeTenantType?

	var inflationRate: BigDecimal?

	var discountRate: BigDecimal?

	val users: List<ObjUser>

	var logoImageId: Any?

	val logoImage: ObjDocument?

}
